feat: реализовать Рейтинги

aigeley opened the pull request:
- добавить пользователю возможность сохранять лайки/дизлайки на события, в которых он принимал участие
- добавить сортировку событий по числу лайков (от большего числа к меньшему)
- добавить возможность получения рекомендаций для текущего пользователя: если пользователь принимал участие в событиях из каких-либо категорий, то ему предлагаются на выбор по одному ближайшему событию из каждой категории, от авторов с наивысшим рейтингом

aigeley committed:
feat: реализовать Рейтинги

- добавить пользователю возможность сохранять лайки/дизлайки на события, в которых он принимал участие
- добавить сортировку событий по числу лайков (от большего числа к меньшему)
- добавить возможность получения рекомендаций для текущего пользователя:
если пользователь принимал участие в событиях из каких-либо категорий, то ему предлагаются на выбор по одному ближайшему событию из каждой категории, от авторов с наивысшим рейтингом

aigeley committed:
fix: Postman-тесты

- парсить число из строки перед сравнением

SquareSweet reviewed (approved):
Оставила пару замечаний, в остальном всё замечательно. :)

aigeley committed:
fix: исправить замечания ревью

- не использовать repository в mapper
- заменить тип данных параметра markValue на Boolean

aigeley commented on ewm-service/src/main/java/ru/practicum/ewm/service/projection/EventWithRequestsMapper.java:
```diff
@@ -1,53 +1,57 @@
 public class EventWithRequestsMapper extends ElementProjectionMapper<Event, EventWithRequests> {
-    private final ParticipationRequestUtils participationRequestUtils;
+    private final EventWithRequestsRepository eventWithRequestsRepository;
 
```
Исправлено

aigeley commented on ewm-service/src/main/resources/schema.sql:
```diff
@@ -66,4 +66,17 @@ CREATE TABLE IF NOT EXISTS events_compilations
+    created    TIMESTAMP DEFAULT CURRENT_TIMESTAMP   NOT NULL,
+    event_id   BIGINT                                NOT NULL,
+    user_id    BIGINT                                NOT NULL,
+    mark_value INTEGER                               NOT NULL,
```
Исправлено
